test/parallel_api/iterator/permutation_iterator.pass.cpp

```diff
@@ -80,6 +80,12 @@ struct is_able_to_to_modify_src_data_in_test<ExecutionPolicy, perm_it_index_tags
     : std::false_type
 {
 };
+
+template <typename ExecutionPolicy>
+struct is_able_to_to_modify_src_data_in_test<ExecutionPolicy, perm_it_index_tags::callable_object>
```

Comment:
Perhaps the changes from #1325 to `is_able_to_to_modify_src_data_in_test` belong better in this PR now that it exists.

Also, I would expect this to be `::std::true_type` rather than `::std::false_type`.  You should be able to use a `callable_object` as an index generator for a `permutation_iterator` on the device and on the host.  Right now a lot of tests are being skipped, so it may make us miss some issues with the result checking.

Similarly, now that I consider it in more detail, I believe that `perm_it_index_tags::transform_iterator` should also be `std::true_type`, assuming that the `transform_iterator` base of the `transform_iterator` is accessible using the `ExecutionPolicy`.


Finally, perhaps we can correct the typo  `is_able_to_to_modify_src_data_in_test` -> `is_able_to_modify_src_data_in_test` here.

Reply:
Partially fixed

Reply:
I have confirmed that #1325 fixes the compiler issues.  As discussed offline, we can leave this as `std::false_type`, and switch it to `std::true_type` in #1325.

Reply:
Also generally, we can wait to adjust the `is_able_to_modify_src_data_in_test` until #1325 or later, and deal with any required fixes at that time.